tests/system-tests/rdscore/internal/rdscorecommon/sriov-pod-level-bond.go

```diff
@@ -0,0 +1,1197 @@
+func VerifyPodLevelBondWorkloadsOnDifferentNodesSamePF() {
+	prepareSecondPodLevelBondDeployment(false, true)
+
+	verifyConnectivity()
+}
+
+// VerifyPodLevelBondWorkloadsOnDifferentNodesDifferentPFs verifies TCP traffic works on the
+// different nodes and different PFs.
+func VerifyPodLevelBondWorkloadsOnDifferentNodesDifferentPFs() {
+	prepareSecondPodLevelBondDeployment(false, false)
+
+	verifyConnectivity()
+}
+
+// VerifyPodLevelBondWorkloadsAfterVFFailOver verifies TCP traffic after bond active interface failure
+// (fail-over procedure).
+//
+//nolint:funlen
+func VerifyPodLevelBondWorkloadsAfterVFFailOver() {
+	prepareSecondPodLevelBondDeployment(false, true)
+
+	verifyConnectivity()
+
+	By("Retrieve client pod-level bond pod object")
+
+	clientPodObj, err := getPodObjectByNamePattern(
+		APIClient,
+		RDSCoreConfig.PodLevelBondDeploymentOneName,
+		RDSCoreConfig.PodLevelBondNamespace)
+	Expect(err).ToNot(HaveOccurred(),
+		fmt.Sprintf("Failed to retrieve client pod-level bond %s object from namespace %s: %v",
+			RDSCoreConfig.PodLevelBondDeploymentOneName, RDSCoreConfig.PodLevelBondNamespace, err))
+
+	By("Retrieve server pod-level bond pod object")
+
+	serverPodObj, err := getPodObjectByNamePattern(
+		APIClient,
+		RDSCoreConfig.PodLevelBondDeploymentTwoName,
+		RDSCoreConfig.PodLevelBondNamespace)
+	Expect(err).ToNot(HaveOccurred(),
+		fmt.Sprintf("Failed to retrieve server pod-level bond %s object from namespace %s: %v",
+			RDSCoreConfig.PodLevelBondDeploymentOneName, RDSCoreConfig.PodLevelBondNamespace, err))
+
+	activeInf, err := getBondActiveInterface(serverPodObj)
+	Expect(err).ToNot(HaveOccurred(),
+		fmt.Sprintf("Failed to retrieve bond active interface for the pod deployment %s in namespace %s: %v",
+			serverPodObj.Definition.Name, serverPodObj.Definition.Namespace, err))
+
+	glog.V(100).Infof("DEBUG LOG: active interface found - %s", activeInf)
+
+	By("Send data from the client container to the IPv4 address used by the server container")
+
+	go func() {
+		output, err := generateTCPTraffic(
+			clientPodObj,
+			RDSCoreConfig.PodLevelBondDeploymentTwoIPv4,
+			RDSCoreConfig.PodLevelBondPort,
+			"10",
+			"5")
+		Expect(err).ToNot(HaveOccurred(),
+			fmt.Sprintf("Failed to generate TCP traffic from the pod %s in namespace %s to the server %s: %v",
+				clientPodObj.Definition.Name, clientPodObj.Definition.Namespace,
+				RDSCoreConfig.PodLevelBondDeploymentTwoIPv4, err))
+
+		testPassed, err := scanClientPodTrafficOutput(output)
+		Expect(err).ToNot(HaveOccurred(),
+			fmt.Sprintf("Failed to parse client pod %s from namespace %s output: %v",
+				clientPodObj.Definition.Name, clientPodObj.Definition.Namespace, err))
+		Expect(testPassed).To(Equal(true),
+			fmt.Sprintf("TCP traffic test verification failed for the pod %s in namespace %s; output %s",
+				clientPodObj.Definition.Name, clientPodObj.Definition.Namespace, output))
+
+		glog.V(100).Infof("DEBUG LOG 222")
```

Comment:
please change to smith more meaningful :)

Reply:
sorry, leftovers, removing :)